django/db/models/sql/query.py

```diff
@@ -1369,14 +1369,14 @@ def build_lookup(self, lookups, lhs, rhs):
         # __exact is the default lookup if one isn't given.
         *transforms, lookup_name = lookups or ["exact"]
         for name in transforms:
-            lhs = self.try_transform(lhs, name)
+            lhs = self.try_transform(lhs, name, lookups)
```

Comment:
This one appears to be key and is covered by tests 👍 

All the others I can update from `.try_transform(lhs, lookup_name, lookups)` to `.try_transform(lhs, lookup_name, [lookup_name])` without any test failures

This makes me think, are we missing some tests?
If the answer to that question is "no, it can only fail here" then maybe we shouldn't force three arguments to `try_transform` and instead have something like...
```python
    def try_transform(self, lhs, name, lookups=None):
         # ....
            if lookups is not None:
                name_index = lookups.index(name)
                unsupported_lookup = LOOKUP_SEP.join(lookups[name_index:])
            else:
                unsupported_lookup = name
            raise FieldError(
                "Unsupported lookup '%s' for %s or join on the field not "
                "permitted%s" % (unsupported_lookup, output_field.__name__, suggestion)
            )
```
so passing in lookups is optional 🤔

Reply:
Hi @sarahboyce. Thank you for your feedback 🙏🏻

I have confirmed that the Lookup error occurs only here.
Therefore, based on your feedback, I have modified the `try_transform` method to make passing `lookups` optional.
Also added the missing tests.